packaging/linux/run_keybase

```diff
@@ -64,8 +66,15 @@ kill_all() {
   if killall Keybase &> /dev/null ; then
     echo Shutting down Keybase GUI...
   fi
+  # In case the package upgrade wasn't able to unmount and remove /keybase
   if fusermount -uz /keybase &> /dev/null ; then
-    echo Unmounting /keybase...
+    # Remove any existing legacy mount.
+    echo Unmounting /keybase and converting to symlink...
+    rmdir /keybase
```

Comment:
Does `rmdir` fail if unmounting failed weirdly that leaves a mountpoint there which can't read anything (i.e. error on read with empty response)?

Also, if it's a fresh install, we are not hitting this codepath?

Reply:
> Does rmdir fail if unmounting failed weirdly that leaves a mountpoint there which can't read anything (i.e. error on read with empty response)?

No idea.  I don't know if `fusermount -uz` can fail in such a way.  If it does, then I guess `/keybase` will be stuck until the next `run_keybase` call.  Any suggestions for changes?

> Also, if it's a fresh install, we are not hitting this codepath?

Correct.

Reply:
> Any suggestions for changes?

Not sure I can think of a clean solution, but I was thinking we can also check all mount points (or simply output from `mount`) and make sure `/keybase` isn't in there. Then `rmdir` just needs to make sure `/keybase` is empty which it already does.

Reply:
I'm confused about what problem this would solve.  I thought you were trying to fix the case where it _isn't_ unmounted correctly, and so would still be listed in the `mount` output.  What exactly are you proposing should happen in the cases where it is and isn't in the `mount` output?

Reply:
Sorry for the confusion! I was thinking about a specific scenario where we have a abnormal kbfsfuse running, and it's not responding to unmount request, and somehow `fusermount -uz` failed to remove the mount point. In this case reading from `/keybase` would get an I/O error (or timeout), but I was not sure whether `rmdir` would error out in that case.

Checking if `/keybase` show up in `mount` is so that we only move on if `/keybase` doesn't show up.

Reply:
Oh you want to exit early if `/keybase` does still show up in `mount`?  I'm not sure I agree with that.  It won't really cause harm and the user will see any stderr output from the `fusermount` and `rmdir` commands on their screens.  That seems reasonable to me but I might still be missing something.

Reply:
> Oh you want to exit early if /keybase does still show up in mount?

Yes that's what I meant.

> It won't really cause harm and the user will see any stderr output from the fusermount and rmdir commands on their screens. That seems reasonable to me but I might still be missing something.

Good point that stderr would make user aware of it. I'm OK with the existing way then. Thanks!